common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/GradleCommand.xtend

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.dsl.common.util
+
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+import java.util.List
+import javax.inject.Inject
+
+class GradleCommand {
+
+	@Inject OSUtil osUtil;
+
+	def String execute(File gradleBuildDirectory, String... commands) {
```

Comment:
We don't need to run an external process here, the Gradle Java API is quite good.
See `org.testeditor.dsl.common.ui.gradle.GradleHelper`.

You can directly access the underlying model using a Java API as explained here: https://gradle.org/blog/feature-spotlight-gradle-tooling-api/